theano/sandbox/rng_mrg.py

```diff
@@ -804,101 +803,125 @@ def pretty_return(self, node_rstate, new_rstate, sample, size, nstreams):
         return sample
 
-    def uniform(self, size, low=0.0, high=1.0, ndim=None, dtype=None,
-                nstreams=None, **kwargs):
-        # TODO : need description for parameter 'size', 'ndim', 'nstreams'
+    def uniform(self, size, low=0.0, high=1.0,
+                ndim=None, dtype=None, nstreams=None, target=None):
+        # TODO: documentation target parameter
         """
-        Sample a tensor of given size whose element from a uniform
-        distribution between low and high.
-
-        If the size argument is ambiguous on the number of dimensions,
-        ndim may be a plain integer to supplement the missing information.
+        Sample a tensor of values from a uniform distribution.
 
         Parameters
         ----------
-        low
-            Lower bound of the interval on which values are sampled.
-            If the ``dtype`` arg is provided, ``low`` will be cast into
-            dtype. This bound is excluded.
-        high
-            Higher bound of the interval on which values are sampled.
-            If the ``dtype`` arg is provided, ``high`` will be cast into
-            dtype. This bound is excluded.
-        size
-          Can be a list of integer or Theano variable (ex: the shape
-          of other Theano Variable).
-        dtype
-            The output data type. If dtype is not specified, it will be
-            inferred from the dtype of low and high, but will be at
-            least as precise as floatX.
+        size : int_vector_like
+            Array dimensions for the output tensor.
+        low : float_like, optional
+            Lower bound of the interval to sample from. This bound is excluded.
+        high : float_like, optional
+            Upper bound of the interval to sample from. This bound is excluded.
+        ndim : int, optional
+            The number of dimensions for the output tensor (defaults to None).
+            This argument is necessary if the size argument is ambiguous on the number of dimensions.
+        dtype : str, optional
+            The data-type for the output tensor. If not specified,
+            the dtype is inferred from low and high, but it is at least as precise as floatX.
+        nstreams : int, optional
+            Number of streams to use for generating random numbers.
+
+        Returns
+        -------
+        samples : TensorVariable
+            A Theano tensor of samples randomly drawn from a uniform distribution.
 
         """
-        low = as_tensor_variable(low)
-        high = as_tensor_variable(high)
+        size = _check_size(size)
+        low = undefined_grad(as_tensor_variable(low))
+        high = undefined_grad(as_tensor_variable(high))
 
         if dtype is None:
-            dtype = scal.upcast(config.floatX, low.dtype, high.dtype)
+            dtype = scal.upcast(floatX, low.dtype, high.dtype)
 
-        low = cast(low, dtype=dtype)
-        high = cast(high, dtype=dtype)
+        low = tensor.cast(low, dtype=dtype)
```

Comment:
This change the graph. Now the undefined_grad is swapped comparted to cast in the graph. I don't think this is a problem, but I prefer to raise it in case someone think it is a problem.

Reply:
This change was introduced to improve the consistency of the code. I used the original implementation of `MRG_RandomStreams.normal` as a reference. Just to provide some more background on where this comes from.

Reply:
So this confirm it should not be a problem. thanks.